src/server/api/circuits.go

```diff
@@ -95,6 +96,61 @@ func circuitLoadHandler(m interfaces.CircuitStorageInterfaceFactory, c *gin.Cont
 	c.JSON(http.StatusOK, circuit)
 }
 
+func circuitUpdateHandler(m interfaces.CircuitStorageInterfaceFactory, c *gin.Context, userId model.UserId) {
+	storageInterface := m.CreateCircuitStorageInterface()
+
+	circuitId := c.Param("id")
+
+	newCircuit, err := parseCircuitRequestData(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	if circuitId == "0" {
```

Comment:
+1 for "0" circuit Id meaning "create".  That's probably better behavior than "circuit ID not found, let's create a new circuit".